src/exchanges/gdax/GDAXFeed.ts

```diff
@@ -289,27 +306,58 @@ export class GDAXFeed extends ExchangeFeed {
         });
     }
 
-    private mapTicker(ticker: GDAXTickerMessage): StreamMessage {
-        return {
+    private attachSequence(streamMessage: StreamMessage,
+                           feedMessage: any): void {
+        if ((feedMessage as any).sequence) {
+            (streamMessage as any).sourceSequence = (feedMessage as any).sequence;
+        }
+    }
+
+    private mapError(errorMessage: GDAXErrorMessage): ErrorMessage {
+        const msg: ErrorMessage = {
+            type: 'error',
+            time: new Date(),
+            message: errorMessage.message,
+            cause: errorMessage.reason
+        };
+        this.emit('feed-error', msg);
+        return msg;
+    }
+
+    private mapUnknown(unknown: any): UnknownMessage {
+        const time = unknown.time ? new Date(unknown.time) : new Date();
+        const product = unknown.product_id;
+        const sequence = unknown.sequence || (product && this.getSequence(product));
+        const msg: UnknownMessage = {
+            type: 'unknown',
+            time: time,
+            sequence: sequence,
+            productId: product
+        };
+        return msg;
+    }
+
+    private mapTicker(ticker: GDAXTickerMessage): TickerMessage {
+        const msg: TickerMessage = {
             type: 'ticker',
             time: ticker.time ? new Date(ticker.time) : new Date(),
             productId: ticker.product_id,
-            sequence: ticker.sequence,
```

Comment:
The last sequence can be part of the ticker

Reply:
Genrally this needs to be updated once the ticker type lands in gdax-node, with some of these fields being potentially undefined.

Reply:
> it appears none of the other exchanges do

That is a very good point.

Afaik GDAX does not guarantee that timestamps are always linear. That is why we rely on sequences as logical timestamps for the most part for event order. I would prefer to keep them in here, in case someone really depends on them.

Reply:
@fb55 I've made it optional for now so that developers know it may exist. I don't have the time to make fake sequences for the other places. PTAL.

BTW `GDAXFeed#handleMessage()` adds `sequence` and `sourceSequence` properties to the StreamMessage if they exist on the input JSON object.